kentikapi/client_integration_test.go

```diff
@@ -579,6 +606,9 @@ type spySyntheticsServer struct {
 	// requests spied by the server
 	requests []*syntheticspb.GetAgentRequest
 	headers  []metadata.MD
+
+	// handlingDelay specifies the delay applied while handling the request
+	handlingDelay *time.Duration
```

Comment:
Nit: Please put it next to responses, because it is connected with it - the way the server handles the request.

BTW: It does not need to be a pointer, because sleep returns immediately with zero value duration